src/components/CytoscapeGraph/graphs/GraphStyles.ts

```diff
@@ -361,8 +385,24 @@ export class GraphStyles {
       return `<div class="${labelDefault} ${labelBox}" style="${labelStyle}">${icons}${contentSpan}</div>`;
     }
 
-    const contentSpan = `<div class="${contentClasses}" style="${contentStyle}">${contentText}</div>`;
-    return `<div class="${labelDefault}" style="${labelStyle}">${icons}${contentSpan}</div>`;
+    let hosts: string[] = [];
+    node.hasVS?.hostnames?.forEach(h => hosts.push(h === '*' ? '(All hosts)' : h));
+    node.isGateway?.ingressInfo?.hostnames?.forEach(h => hosts.push(h === '*' ? '(All hosts)' : h));
+
+    let htmlHosts = '';
+    if (hosts.length !== 0) {
+      let hostsToShow = hosts;
+      if (hostsToShow.length > config.graph.maxHosts) {
+        hostsToShow = hosts.slice(0, config.graph.maxHosts);
+        hostsToShow.push((hosts.length - config.graph.maxHosts) === 1 ? "(and one more host)" : `(and ${hosts.length - config.graph.maxHosts} more hosts)`);
```

Comment:
maybe ellipses and no parentheses?  Also, maybe only numbers (I'm also OK with just keeping 'one')?

1 more host...
5 more hosts...

Reply:
OK, looks cleaner, I'll use your provided example.